lib/auth/webauthnwin/webauthn_windows.go

```diff
@@ -0,0 +1,603 @@
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	opts, err := n.assertOptionsToCType(in.Response, loginOpts)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	var out *webauthnAssertion
+	ret, _, lastErr := procWebAuthNAuthenticatorGetAssertion.Call(
+		uintptr(hwnd),
+		uintptr(unsafe.Pointer(rpid)),
+		uintptr(unsafe.Pointer(cd)),
+		uintptr(unsafe.Pointer(opts)),
+		uintptr(unsafe.Pointer(&out)),
+	)
+	if ret != 0 {
+		return nil, trace.Wrap(getErrorNameOrLastErr(ret, lastErr))
+	}
+	if out == nil {
+		return nil, errors.New("unexpected nil response from GetAssertion")
+	}
+
+	// Note that we need to copy bytes out of `out` if we want to free object.
+	// That's why bytesFromCBytes is used.
+	// We don't care about free error so ignore it explicitly.
+	defer func() { _ = freeAssertion(out) }()
+
+	authData := bytesFromCBytes(out.cbAuthenticatorData, out.pbAuthenticatorData)
+	signature := bytesFromCBytes(out.cbSignature, out.pbSignature)
+	userID := bytesFromCBytes(out.cbUserId, out.pbUserId)
+	credential := bytesFromCBytes(out.Credential.cbId, out.Credential.pbId)
+	credType := windows.UTF16PtrToString(out.Credential.pwszCredentialType)
+
+	return &wanlib.CredentialAssertionResponse{
+		PublicKeyCredential: wanlib.PublicKeyCredential{
+			RawID: credential,
+			Credential: wanlib.Credential{
+				ID:   base64.RawURLEncoding.EncodeToString(credential),
+				Type: credType,
+			},
+		},
+		AssertionResponse: wanlib.AuthenticatorAssertionResponse{
+			AuthenticatorData: authData,
+			Signature:         signature,
+			UserHandle:        userID,
+			AuthenticatorResponse: wanlib.AuthenticatorResponse{
+				ClientDataJSON: jsonEncodedCD,
+			},
+		},
+	}, nil
+}
+
+// MakeCredential calls WebAuthNAuthenticatorMakeCredential endpoint from
+// webauthn.dll and returns CredentialCreationResponse.
+// It interacts with both FIDO2 and Windows Hello depending on
+// wanlib.CredentialCreation (using auto starts with Windows Hello but there is
+// option to select other devices).
+// Windows Hello keys are always resident.
+func (n *nativeImpl) MakeCredential(origin string, in *wanlib.CredentialCreation) (*wanlib.CredentialCreationResponse, error) {
+	hwnd, err := getForegroundWindow()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	rp, err := rpToCType(in.Response.RelyingParty)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	u, err := userToCType(in.Response.User)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	credParam, err := credParamToCType(in.Response.Parameters)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	cd, jsonEncodedCD, err := clientDataToCType(in.Response.Challenge.String(), origin, string(protocol.CreateCeremony))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	opts, err := n.makeCredOptionsToCType(in.Response)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	var out *webauthnCredentialAttestation
+	ret, _, lastErr := procWebAuthNAuthenticatorMakeCredential.Call(
```

Comment:
nit:

```suggestion
	ret, _, err := procWebAuthNAuthenticatorMakeCredential.Call(
```

I'm not seeing a reason to name this one error differently?

Reply:
i have named it differently to indicate that this is non standard error. For example you cannot use:
```
if err!=nil {
   return nil
}
```
because if everything went fine inside that value there will be error equal to `This operation finished successful` or something similar. 
> // The returned error is always non-nil, constructed from the result of GetLastError.
// Callers must inspect the primary return value to decide whether an error occurred
// (according to the semantics of the specific function being called) before consulting
// the error. The error always has type syscall.Errno.


I don't have strong preferences and can revert it back to err.

Other option is ignore it completely which  is done in few libraries: https://github.com/AllenDang/w32/blob/ad0a36d80adcd081d5c0dded8e97a009b486d1db/user32.go#L219-L226
i think ignoring it also makes some sense. I have actually never received any error there.
If API fails with something - then we got non 0 error which we translate using another API call.
When something is mixed with input types - it results in panic.

Reply:
I think it's important to maintain a consistent idiom, regardless of the choice we make. I'm OK ignoring the err if you think that's best - up to you.

Reply:
ok, I will keep error checks just in case but will rename it to `err`.